dulwich/tests/test_client.py

```diff
@@ -408,6 +408,12 @@ def test_http(self):
         self.assertTrue(isinstance(c, HttpGitClient))
         self.assertEqual('/jelmer/dulwich', path)
 
+    def test_local_specified_cls(self):
+        c, path = get_transport_and_path(
+            'foo.bar/baz', local_git_client_cls=SubprocessGitClient)
```

Comment:
No. This test passes because we are just asserting on the type of client returned. It never calls any methods on the clients, so the client never calls git.

This commit actually fixes a bunch of tests from `dulwich.tests.test_porcelain` that were failing when git was not installed because the default client was `SubprocessGitClient`, and we were not checking if it was installed with `require_git_version`.